datastream-cloud-storage/src/main/java/com/linkedin/datastream/cloud/storage/io/ScribeParquetAvroConverter.java

```diff
@@ -0,0 +1,336 @@
+package com.linkedin.datastream.cloud.storage.io;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.apache.avro.Schema;
+import org.apache.avro.SchemaBuilder;
+import org.apache.avro.generic.GenericArray;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * ScribeParquetAvroConverter handles the conversion of avro to parquet compatible avro schema
+ */
+public class ScribeParquetAvroConverter {
+  private static final Logger LOG = LoggerFactory.getLogger(ScribeParquetAvroConverter.class);
+  public static final String SCRIBE_HEADER = "scribeHeader";
+
+  /**
+   * Converts field avro schema to parquet compatible format and handles special cases like timestamp, Arrays, Records.
+   * @param isNotNullable  Whether the field is required or not
+   * @param fieldSchema    Schema of the field
+   * @param field          Field
+   * @param fieldAssembler FieldAssembler
+   * @return SchemaBuilder.FieldAssembler<Schema>
+   */
+  public static SchemaBuilder.FieldAssembler<Schema> getFieldAssembler(Boolean isNotNullable, Schema fieldSchema, Schema.Field field, SchemaBuilder.FieldAssembler<Schema> fieldAssembler) {
+    try {
+      String typeName = fieldSchema.getType().name();
+      if (typeName.equalsIgnoreCase("ARRAY")) {
+        // For list of nested objects
+        if (fieldSchema.getElementType().getType().getName().equalsIgnoreCase("RECORD")) {
+          Schema elementTypeSchema = generateParquetStructuredAvroSchema(fieldSchema.getElementType());
+          fieldAssembler.name(field.name().toLowerCase())
+              .doc(field.doc())
+              .type()
+              .optional()
+              .array().items(elementTypeSchema);
+        } else {
+          getFieldAssemblerForRecord(isNotNullable, fieldSchema, field, fieldAssembler);
+        }
+      } else if (typeName.equalsIgnoreCase("MAP")) {
+        getFieldAssemblerForRecord(isNotNullable, fieldSchema, field, fieldAssembler);
+      } else if (typeName.equalsIgnoreCase("RECORD")) {
+        if (fieldSchema.getName().equalsIgnoreCase("scribe_header")) {
```

Comment:
`scribe_header ` is fieldSchema name for `scribeHeader` field. Since all our event names are in snake_case, the nested objects are also named in snake_case and in this case we are checking with 
[here](http://kube-kafka-schema-c1.service.intrabo1.consul.csnzoo.com/subjects/scribe.v2.nestedobjects.scribe_internal.scribe_header/versions/latest) is schema